account/manager.go

```diff
@@ -407,6 +417,28 @@ func (m *Manager) maybeBroadcastTx(ctx context.Context, tx *wire.MsgTx,
 	return m.cfg.Wallet.PublishTransaction(ctx, tx, label)
 }
 
+// verifyAccountSigner ensures that we are able to recreate the account
+// secret for active accounts. That means that the LND signerClient did
+// not change and we are able to generate valid signatures for this account.
+func (m *Manager) verifyAccountSigner(ctx context.Context,
+	account *Account) error {
+
+	// The secret was based on both base keys, the trader and auctioneer's.
+	secret, err := m.cfg.Signer.DeriveSharedKey(
+		ctx, account.AuctioneerKey, &account.TraderKey.KeyLocator,
+	)
+	if err != nil {
+		return fmt.Errorf("unable to regenerate secret: %v", err)
+	}
+
+	// Here we would detect if the backend LND node (signer) changed.
+	if !bytes.Equal(secret[:], account.Secret[:]) {
+		return fmt.Errorf("unable to verify secret")
```

Comment:
This error message could probably be a bit more descriptive. For example: "couldn't derive account secret; make sure you are using the same lnd node/seed that was used for creating the account".